include/config/config_safeguards.h

```diff
@@ -0,0 +1,146 @@
+#pragma once
+
+/*****************************
+ * COMPATIBILITY SAFE GUARDS *
+ *****************************/
+
+/**
+ * NOTE: Only mess with these if you know what you're doing!
+ * These are put in place to insure that connected functionality works as intended.
+ */
+
+
+/*****************
+ * config_graphics
+ */
+
+#ifndef F3DZEX_GBI_2
+    #undef OBJECTS_REJ // OBJECTS_REJ requires f3dzex.
+#endif // !F3DZEX_GBI_2
+
+#ifndef F3DEX_GBI_SHARED
+    #undef OBJECTS_REJ // Non F3DEX-based ucodes do NOT support ucode switching.
+#endif // !F3DEX_GBI_SHARED
+
+#ifdef OBJECTS_REJ
+// Enable required ucodes.
+    #define F3DEX2_REJ_GBI
+    #define F3DLX2_REJ_GBI
+#endif // OBJECTS_REJ
+
+
+/*****************
+ * config_debug
+ */
+
+#ifdef PUPPYPRINT_DEBUG
+    #ifndef PUPPYPRINT
+        #define PUPPYPRINT
+    #endif
+#endif
+
+#ifdef COMPLETE_SAVE_FILE
+    #define UNLOCK_ALL
+#endif // COMPLETE_SAVE_FILE
+
+#ifdef DEBUG_ALL
+    #define DEBUG_LEVEL_SELECT
+    #define ENABLE_DEBUG_FREE_MOVE
+    #define PUPPYPRINT
+    #define PUPPYPRINT_DEBUG 1
+    #define VISUAL_DEBUG
+    #define UNLOCK_ALL
+    #define COMPLETE_SAVE_FILE
+#endif // DEBUG_ALL
```

Comment:
If I'm not mistaken, this can also lead to compiler warnings via duplicate defines.